provider/service_data.go

```diff
@@ -132,22 +120,29 @@ func dataFireHydrantServices(ctx context.Context, d *schema.ResourceData, m inte
 
 	services := make([]interface{}, 0)
 
+	// Set the data source attributes to the values we got from the API
 	for _, svc := range r.Services {
 		values := map[string]interface{}{
 			"id":           svc.ID,
-			"name":         svc.Name,
+			"alert_on_add": svc.AlertOnAdd,
 			"description":  svc.Description,
+			"name":         svc.Name,
 			"service_tier": svc.ServiceTier,
-			"alert_on_add": svc.AlertOnAdd,
 		}
+
+		// Process any attributes that could be nil
+		if svc.Owner != nil {
+			values["owner_id"] = svc.Owner.ID
+		}
```

Comment:
It feels like this `map[string]interface{}` assignment could be abstracted into a helper function that takes in a `Service` struct and assigns these values. Reason being is the singular data source and list data source have the same logic.